### PR TITLE
Add basic BATS test suite for freight-{add,cache}

### LINKED ISSUE #5: Test suite required
A test suite is required to help us prevent regressions, something that can easily be enabled with Travis CI on this repository.

As to frameworks, [bats](https://github.com/sstephenson/bats#readme) would be quite an easy way to write tests in the same language as the project.  It can be installed easily on Travis CI, and we just create a directory of .bats files containing individual tests to test different pieces of functionality.

### PR DESCRIPTION
**Pull request by domcleal**
A set of tests for freight-add and freight-cache are added to check
that an ordinary deb can be added and generated into a repository.
A further test runs apt locally against the Freight cache to check the
repo is functional.

Fixes #5 

---

@mattock you'll need to enable Travis CI on the repo I think.  The tests will also fail at the moment due to #8.

### CODE REVIEW

**Comment by mmoll** [comment]
this is magic :rainbow:
:+1:

**Comment by mattock** [comment]
@domcleal: I will look into Travis (and all the other stuff) tomorrow.

**Comment by domcleal** [comment]
I've added an extra test for the multiple GPG key support, and also added a matrix to the Travis CI config to run tests under both dash and bash (though called as 'sh' for POSIX mode).

File: bin/freight [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/env sh
```
Comment (mmoll) [inline bin/freight]: this is `/usr/bin/env` on all my systems (Debian and FreeBSD)

**Comment by mmoll** [comment]
One comment, working perfectly fine otherwise

File: bin/freight (hunk 1 above)
Comment (domcleal) [inline bin/freight]: Thanks, fixed.

**Comment by mmoll** [comment]
:+1:

**Comment by mmoll** [comment]
thanks @domcleal!

**Pull request merged**